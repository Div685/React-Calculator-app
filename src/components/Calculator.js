import React from 'react';
import App from './App';
import '../style/Calculator.css';

function Calculator() {
  return (
    <div className="main-calculator">
      <h1>Let&apos;s do some math!</h1>
      <App />
    </div>
  );
}

export default Calculator;
