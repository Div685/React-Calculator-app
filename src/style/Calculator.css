.main-calculator {
  display: flex;
}

.main-calculator h1 {
  font-family: 'Times New Roman', Times, serif;
  margin: 25px auto;
}
