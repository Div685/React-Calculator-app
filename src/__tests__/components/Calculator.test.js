import React from 'react';
import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';

import Calculator from '../../components/Calculator';

describe('Calculator component', () => {
  it('renders calculator page correctly', () => {
    const tree = renderer.create(<Calculator />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('Test for calculator component', () => {
  it('test the heading of calculator component', () => {
    render(<Calculator />);
    expect(screen.getByRole('heading')).toHaveTextContent('Let\'s do some math!');
  });
});
