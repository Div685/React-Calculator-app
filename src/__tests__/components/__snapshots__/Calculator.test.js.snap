// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calculator component renders calculator page correctly 1`] = `
<div
  className="main-calculator"
>
  <h1>
    Let's do some math!
  </h1>
  <div
    className="main-calc"
  >
    <div
      className="display"
    >
      <span>
        0
      </span>
    </div>
    <div
      className="main-div"
    >
      <div
        className="group-row"
      >
        <button
          className={null}
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": null,
              "width": "25%",
            }
          }
          type="button"
        >
          AC
        </button>
        <button
          className={null}
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": null,
              "width": "25%",
            }
          }
          type="button"
        >
          +/-
        </button>
        <button
          className={null}
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": null,
              "width": "25%",
            }
          }
          type="button"
        >
          %
        </button>
        <button
          className="#F5913E"
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "#F5913E",
              "width": "25%",
            }
          }
          type="button"
        >
          รท
        </button>
      </div>
      <div
        className="group-row"
      >
        <button
          className={null}
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": null,
              "width": "25%",
            }
          }
          type="button"
        >
          7
        </button>
        <button
          className={null}
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": null,
              "width": "25%",
            }
          }
          type="button"
        >
          8
        </button>
        <button
          className={null}
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": null,
              "width": "25%",
            }
          }
          type="button"
        >
          9
        </button>
        <button
          className="#F5913E"
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "#F5913E",
              "width": "25%",
            }
          }
          type="button"
        >
          x
        </button>
      </div>
      <div
        className="group-row"
      >
        <button
          className={null}
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": null,
              "width": "25%",
            }
          }
          type="button"
        >
          4
        </button>
        <button
          className={null}
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": null,
              "width": "25%",
            }
          }
          type="button"
        >
          5
        </button>
        <button
          className={null}
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": null,
              "width": "25%",
            }
          }
          type="button"
        >
          6
        </button>
        <button
          className="#F5913E"
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "#F5913E",
              "width": "25%",
            }
          }
          type="button"
        >
          -
        </button>
      </div>
      <div
        className="group-row"
      >
        <button
          className={null}
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": null,
              "width": "25%",
            }
          }
          type="button"
        >
          1
        </button>
        <button
          className={null}
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": null,
              "width": "25%",
            }
          }
          type="button"
        >
          2
        </button>
        <button
          className={null}
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": null,
              "width": "25%",
            }
          }
          type="button"
        >
          3
        </button>
        <button
          className="#F5913E"
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "#F5913E",
              "width": "25%",
            }
          }
          type="button"
        >
          +
        </button>
      </div>
      <div
        className="group-row"
      >
        <button
          className={null}
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": null,
              "width": "50%",
            }
          }
          type="button"
        >
          0
        </button>
        <button
          className={null}
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": null,
              "width": "25%",
            }
          }
          type="button"
        >
          .
        </button>
        <button
          className="#F5913E"
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "#F5913E",
              "width": "25%",
            }
          }
          type="button"
        >
          =
        </button>
      </div>
    </div>
  </div>
</div>
`;
